tritium-processor/src/main/java/com/palantir/tritium/processor/ProxyAnnotationProcessor.java

```diff
@@ -0,0 +1,426 @@
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.ExecutableType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.type.TypeVariable;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.SimpleElementVisitor8;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic.Kind;
+
+public final class ProxyAnnotationProcessor extends AbstractProcessor {
+
+    private static final String HANDLER_NAME = "handler";
+    private static final ImmutableSet<String> ANNOTATIONS = ImmutableSet.of(Proxy.class.getName());
+
+    private final Set<Name> invalidElements = new HashSet<>();
+    private Messager messager;
+    private Filer filer;
+    private Elements elements;
+    private Types types;
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
+        super.init(processingEnv);
+        this.messager = processingEnv.getMessager();
+        this.filer = processingEnv.getFiler();
+        this.elements = processingEnv.getElementUtils();
+        this.types = processingEnv.getTypeUtils();
+    }
+
+    @Override
+    public SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latestSupported();
+    }
+
+    @Override
+    public ImmutableSet<String> getSupportedAnnotationTypes() {
+        return ANNOTATIONS;
+    }
+
+    @Override
+    public boolean process(Set<? extends TypeElement> _annotations, RoundEnvironment roundEnv) {
+        if (roundEnv.processingOver()) {
+            if (!invalidElements.isEmpty()) {
+                messager.printMessage(Kind.ERROR, "Processing completed with unresolved elements: " + invalidElements);
+            }
+            return false;
+        }
+        for (Element element : getElementsToProcess(roundEnv)) {
+            if (element.getKind() != ElementKind.INTERFACE) {
+                messager.printMessage(
+                        Kind.ERROR,
+                        "Only interfaces may be instrumented using @" + Proxy.class.getSimpleName(),
+                        element);
+                continue;
+            }
+            TypeElement typeElement = (TypeElement) element;
+            List<DeclaredType> allInterfaces = new ArrayList<>();
+            List<DeclaredType> minimalInterfaces = new ArrayList<>();
+            if (isInvalid(element.asType(), allInterfaces, minimalInterfaces)) {
+                invalidElements.add(typeElement.getQualifiedName());
+                continue;
+            }
+            if (allInterfaces.isEmpty()) {
+                messager.printMessage(
+                        Kind.ERROR,
+                        "Cannot generate a instrumented implementation. "
```

Comment:
```suggestion
                        "Cannot generate an instrumented implementation. "
```

Reply:
I probably need to fix this in the tritium processor as well. I need to audit for the word `instrument` and replace with `proxy.